script.js

```diff
@@ -1,49 +1,82 @@
-const db_local = 'stays.json'
-const db_url = 'https://raw.githubusercontent.com/pamdayne/wind-d-bnb/main/db/stays.json'
+const DB_URL = "//raw.githubusercontent.com/pamdayne/wind-d-bnb/main/db/stays.json";
 
 async function fetchLocations() {
-	try {
-		let resp = await fetch(db_url)
-		return await resp.json()
-	} catch (err) {
-		console.error(err)
-	}
+  return fetch(DB_URL)
+    .then(function (resp) { return resp.json() })
+    .then(function (json) {
+      return json;
+    }).catch(function (error) {
+      console.error(error);
+    })
 }
 
+// HOW CAN WE WRITE THIS BETTER? :D Anymore I can improve on?
 async function renderList() {
-	let places = await fetchLocations()
-	let html = ''
-
-	places.forEach(place => {
-		setCountry(place.country)
-
-		html += `<div class="lists-wrapper">
-								<div class="photo">
-									<img
-										src="${place.photo}"
-										alt="${place.title}" />
-								</div>
-								<div class="details">
-									<div class="unit">
-									${place.superHost != false ? '<div class="host"> Super Host</div>' : ''}
-										<div class="type">${place.type}</div >
-										<div class="bed">
-										${place.beds != null ? '<span class="total">' + place.beds + ' beds</span > ' : ''}
-										</div >
-										<div class="score">
-										<span class="material-icons">star</span>
-										<span class="rating">${place.rating}</span>
-										</div>
-									</div>
-									<div class="short-desc">${place.title}</div>
-								</div >
-							</div > `
-		document.querySelector('#locations').innerHTML = html
-	})
+  let places = await fetchLocations()
+  let featImg, typeDiv, hostDiv, rateDiv, bedDiv, descDiv, unitDiv, detailsDiv, mainDiv = {}
+
+  places.forEach(location => {
+    const { photo, title, superHost, type, beds, rating } = location;
+
+    // The place's image
+    featImg = setFeaturedImage(photo, title, 'photo')
+
+    // Unit's details
+    hostDiv = superHost ? setContentElement('div', 'SUPER HOST', 'host') : ''
```

Comment:
Discussed with Nipuna:
- Suggest to use `null` instead of `''` because it is not a string.